Code/Tools/ProjectManager/Source/GemCatalog/GemCatalogScreen.cpp

```diff
@@ -144,11 +149,80 @@ namespace O3DE::ProjectManager
                 {
                     m_gemModel->AddGem(gemInfoResult.GetValue<GemInfo>());
                     m_gemModel->UpdateGemDependencies();
+                    m_proxyModel->sort(/*column=*/0);
                 }
             }
         }
     }
 
+    void GemCatalogScreen::Refresh(const QString& projectPath)
+    {
+        QHash<QString, GemInfo> gemInfoHash;
+
+        // create a hash with the gem name as key
+        AZ::Outcome<QVector<GemInfo>, AZStd::string> allGemInfosResult = PythonBindingsInterface::Get()->GetAllGemInfos(projectPath);
+        if (allGemInfosResult.IsSuccess())
+        {
+            QVector<GemInfo> gemInfos = allGemInfosResult.GetValue();
```

Comment:
Looks like GetValue returns a reference so we could avoid copying a potentially large vector! Do you think you could also change the other uses of this GetAllGemInfos to do that?

Reply:
good catch!